src/warc2zim/utils.py

```diff
@@ -41,7 +41,7 @@ def parse_title(content):
 
 def to_string(input: str | bytes) -> str:
     try:
-        input = input.decode("utf8")
+        input = input.decode("utf-8-sig")
```

Comment:
Sadly, we have to handle the BOM anyway as it may be present in the content we get:

```python
import requests

css = requests.get("https://donorbox.org/assets/application_embed-47da8f7456acb6aa58b61f2e5c664fccbf3cae5b0ad587f129dcd2d93caa65e8.css").content

print(content[:20])
```

Reply:
For this particular use case it doesn't matter (it's just a zero width space) but for URLs it could be problematic as a typable url/path would become un-typable.
Doesn't harm anyway...

Reply:
But it break the parsing of the css:

```python
import tinycss2

css = requests.get(...)

# print `<QualifiedRule … { … }>` (skipping the first (at) rule)
print(tinycss2.parse_stylesheet(css.decode('utf-8'))[0])

# print `<AtRule @import … { … }>`
print(tinycss2.parse_stylesheet(css.decode('utf-8-sig'))[0])
```

Reply:
Indeed the first rule doesn't appear in content (but is in prelude and thus serialized – but we wouldn't rewrite it!). Looks like we should open a ticket upstream.
Can you add a brief comment explaining tinycss2 doesn't handle it correctly? Including that sample URL would help I think.

Reply:
Issue opened upstream : https://github.com/Kozea/tinycss2/issues/52

However, I'm  not sure it is a bug in tinycss. It is more us passing a not correctly decoded content to tinycss.

> Can you add a brief comment explaining tinycss2 doesn't handle it correctly? Including that sample URL would help I think.

I will add a link to the tinycss2 issue as a comment.

Reply:
> However, I'm  not sure it is a bug in tinycss. It is more us passing a not correctly decoded content to tinycss.

We'll see what they think about. It will probably come down to what the CSS spec says. I wonder if browsers take care of it before sending it to parser or not.